src/libponyc/type/viewpoint.c

```diff
@@ -350,8 +350,14 @@ static void replace_type(ast_t** astp, ast_t* target, ast_t* with)
           switch(ast_id(eph))
           {
             case TK_EPHEMERAL:
-              a_with = consume_type(with, TK_NONE);
+            {
+              ast_t* c_with = consume_type(with, TK_NONE);
+              if (c_with != NULL)
+              {
+                a_with = c_with;
```

Comment:
this was a_with = with but that is already what it is.

i changed to `= c_with` as I believe that was the intent.